pcm_wav.cpp

```diff
@@ -45,13 +51,13 @@ PcmOutputWav::PcmOutputWav(const char *fn)
     header.nBitsPerSample = 16;
     strncpy(header.tagdata, "data", 4);
     header.datasize = 0;
-
-    f = fopen(fn, "wb");
+    Filename = fn;
+    f = fopen(Filename, "wb");
     if (f == NULL) {
-        perror("fopen");
-        exit(1);
+        perror("failed to open output file.");
```

Comment:
If you don't do something to handle the error here, what happens? You're going to call fwrite with a NULL file pointer.